generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,965 @@
+        Other states are all non applicable since they are invalid to begin with
+
+        So verification would be:
+        if A is deleted and refA is deleted: return False
+        if A is added and refA is added: return False
+        return True
+        """
+        simulated_config = move.apply(diff.current_config)
+        deleted_paths, added_paths = self._get_paths(diff.current_config, simulated_config, [])
+
+        if not self._validate_paths_config(deleted_paths, diff.current_config):
+            return False
+
+        if not self._validate_paths_config(added_paths, diff.target_config):
+            return False
+
+        return True
+
+    def _get_paths(self, current_ptr, target_ptr, tokens):
+        deleted_paths = []
+        added_paths = []
+
+        if isinstance(current_ptr, list) or isinstance(target_ptr, list):
+            tmp_deleted_paths, tmp_added_paths = self._get_list_paths(current_ptr, target_ptr, tokens)
+            deleted_paths.extend(tmp_deleted_paths)
+            added_paths.extend(tmp_added_paths)
+            return deleted_paths, added_paths
+
+        if not isinstance(current_ptr, dict):
+            if current_ptr != target_ptr:
+                # tokens.append(token)
+                deleted_paths.append(self.path_addressing.create_path(tokens))
+                added_paths.append(self.path_addressing.create_path(tokens))
+                # tokens.pop()
+
+            return deleted_paths, added_paths
+
+        for token in current_ptr:
+            tokens.append(token)
+            if token not in target_ptr:
+                deleted_paths.append(self.path_addressing.create_path(tokens))
+            else:
+                tmp_deleted_paths, tmp_added_paths = self._get_paths(current_ptr[token], target_ptr[token], tokens)
+                deleted_paths.extend(tmp_deleted_paths)
+                added_paths.extend(tmp_added_paths)
+            tokens.pop()
+
+        for token in target_ptr:
+            tokens.append(token)
+            if token not in current_ptr:
+                added_paths.append(self.path_addressing.create_path(tokens))
+            tokens.pop()
+
+        return deleted_paths, added_paths
+
+    def _get_list_paths(self, current_list, target_list, tokens):
+        """
+        Gets all paths with the given list, assume list items are unique
+        """
+        deleted_paths = []
+        added_paths = []
+
+        hashed_target = set(target_list)
+        for index, value in enumerate(current_list):
+            if value not in hashed_target:
+                tokens.append(index)
+                deleted_paths.append(self.path_addressing.create_path(tokens))
+                tokens.pop()
+
+        hashed_current = set(current_list)
+        for index, value in enumerate(target_list):
+            if value not in hashed_current:
+                tokens.append(index)
```

Comment:
comment:
`added_paths refer to paths in the target config and not necessarily the current config`